js/views/conversation_view.js

```diff
@@ -343,12 +348,15 @@
             var toast;
             if (extension.expired()) {
                 toast = new Whisper.ExpiredToast();
-                toast.$el.insertAfter(this.$el);
-                toast.render();
-                return;
             }
             if (this.model.isPrivate() && storage.isBlocked(this.model.id)) {
                 toast = new Whisper.BlockedToast();
+            }
+            if (!this.model.isPrivate() && this.model.get('left')) {
+                toast = new Whisper.LeftToast();
+            }
+
+            if (toast) {
```

Comment:
I think this arrangement works. Blocked and Left are mutually exclusive, and the expired build is a global issue that must be resolved before the others in any case.